### PR TITLE
Fix race condition and refactor to simplify install method in mainInstallButton

### LINKED ISSUE #3371: [Featured Experiments] Loading indicator and onboarding not displayed
### STR

1. Make sure test pilot is uninstalled
2. click the `install test pilot and enable notes` button

### Results
Test Pilot and notes install but there is no barberpoll on the installing notes button and the experiment onboarding tour does not appear.

### PR DESCRIPTION
**Pull request by meandavejustice**
- fixes #3371

### CODE REVIEW

File: frontend/src/app/components/MainInstallButton/index.js [hunk 1]
```diff
@@ -26,6 +26,17 @@ export default class MainInstallButton extends React.Component {
     };
   }
 
+  enableExperimentWrapped() {
+    const { experiment, enableExperiment, postInstallCallback } = this.props;
+
+    this.setState({ isInstalling: true }, () => enableExperiment(experiment)
+      .then(() => {
+        if (postInstallCallback) postInstallCallback();
```
Comment (chuckharmston) [inline frontend/src/app/components/MainInstallButton/index.js]: Braces here, please.

**Review by chuckharmston — APPROVED** [review]
Other than one nit, it looks and works well. Thanks!

File: frontend/src/app/components/MainInstallButton/index.js (hunk 1 above)
Comment (meandavejustice) [reply frontend/src/app/components/MainInstallButton/index.js]: You want inline or newlines?

Comment (chuckharmston) [reply frontend/src/app/components/MainInstallButton/index.js]: Newlines.

**Comment by johngruen** [comment]
I am not seeing the fix working here.

feb-21-2018 09-57-02

**Comment by meandavejustice** [comment]
@johngruen Anything special to reproduce?

**Comment by meandavejustice** [comment]
@johngruen This is what I'm getting, maybe try pulling again? I just pushed another commit slightly simplifying the code to make it more readable.

one-click-install

**Comment by chuckharmston** [comment]
I see what Dave is seeing here.

**Comment by johngruen** [comment]
@chuckharmston @meandavejustice check the url, you guys are on /experiments installers will mostly be at / where i am still seeing the issue

**Comment by meandavejustice** [comment]
@johngruen just pointed out to me that the behavior he recorded above happens when the you visit `/` and what me and @chuckharmston were seeing was when you start out on `experiments`. Which is caused by the page updating and switching views mid installAndEnable, once the TXP addon is installed. https://github.com/mozilla/testpilot/blob/master/frontend/src/app/containers/HomePage/index.js#L10

Which led me to look at our home page views and wonder if `homePageWithAddon` and `homePageWithoutAddon` are separate for historical reasons. We may be able to combine them into one view, (separating out appropriate components) and avoid this rerender that messes up our one click installs, and avoids code duplication as well.

**Pull request merged**